src/preamble.js

```diff
@@ -269,6 +269,14 @@ var HEAP_DATA_VIEW;
 var HEAP64;
 #endif
 
+#if USE_PTHREADS
+if (ENVIRONMENT_IS_PTHREAD) {
+  // Grab imports from the pthread to local scope.
+  buffer = Module['buffer'];
```

Comment:
This block was moved from shell_pthreads.js so that the assignment to buffer comes after its declaration.